lib/puppet/module/task.rb

```diff
@@ -61,6 +62,33 @@ def self.is_tasks_filename?(path)
       return true
     end
 
+    def self.find_files(name, metadata)
+      if metadata.key?('files')
+        file_list = metadata['files'].map do |file|
+          splits = file.split("/")
+          pup_module = Puppet::Module.find(splits[0])
+          if pup_module.nil?
+            Puppet.warning(_("Could not find module %{module}" % { module: splits[0] }))
+            return []
+          end
+          mount = splits[1]
+          unless MOUNTS.include? mount
+            msg = _("Files must be saved in module directories that Puppet makes available via mount points: files, lib, tasks")
+            raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+          end
+          path = File.join(pup_module.path, mount, splits[2..-1])
+          unless Pathname.new(path).absolute?
+            msg = _("File pathnames cannot include relative paths")
+            raise InvalidMetadata.new(msg, 'puppet.tasks/invalid-metadata')
+          end
+
+          { "name" => name, "path" =>  path }
```

Comment:
This format is required by the TaskData schema in puppetserver. puppetserver is going to treat `name` that as the "filename", so I think we should put `splits[2..-1]` for the name. We also need to preserve the module name somehow for lookup. So I'd include the whole `file` var.

Reply:
This will also require some changes to puppetserver. We process `files` from Puppet with `describe-task-file` in Puppetserver, which assumes every file comes from the task's module. Since that's no longer the case, the uri we construct will be wrong.